massage/resynth/guitar_strummer.py

```diff
@@ -0,0 +1,275 @@
+        jam : JAMS
+            a jams object that contain 'chord' annotation for y
+        instrument_label : str
+            Label specifying the instrument of the track. Will add more
+                -'acoustic guitar' : Acoustic Guitar
+        """
+
+        # Get Chord annotation
+        time_intervals, chord_labs = jam.search(
+            namespace='chord')[0].to_interval_values()
+
+        chord_sequence = []
+        for interval, lab in zip(time_intervals, chord_labs):
+            chord_sequence.append([interval[0], interval[1], lab])
+
+        envelope = util.get_energy_envelope(y)
+        y_mono = librosa.core.to_mono(y)
+
+        onsets, offsets, energies = util.onset_offset(y=y_mono, sr=fs)
+        velocities = util.amplitude_to_velocity(energies)
+        sound_font, program = util.pick_sf(y_mono, fs, instrument_label)
+
+        # print("soundfont: {} program: {}".format(sound_font, program))
+
+        env_times = librosa.samples_to_time(np.arange(len(envelope[0])), sr=fs)
+
+        energy_interp = []
+        # list of interp1d objects, len(.) = number of channels
+        for ch in range(envelope.shape[0]):
+            energy_interp_ch = scipy.interpolate.interp1d(
+                env_times, envelope[0], bounds_error=False,
+                fill_value='extrapolate')
+            energy_interp.append(energy_interp_ch)
+
+        # Generating midi for synthesis and also jams file
+        midi_data = self._generate_chord_midi(
+            chord_sequence, onsets, offsets, velocities, program,
+            energy_interp, VOICING_FILE)
+        # Convert midi_data -> jams file
+        jams_out = util.midi_to_jams(midi_data) # audio duration
+
+        # Resynth
+        y = midi_data.fluidsynth(sf2_path=sound_font, fs=fs)
+        # np.savez('mock_y', y)
+        if len(y) == 0:
+            print("y is none")
+            return
+
+        # Normalize
+        if np.max(np.abs(y)) > 0:
+            y = y / np.max(np.abs(y))
+
+        y_len = len(y)
+        env_len = len(envelope[0])
+        if y_len > env_len:
+            y_stereo = np.array(
+                [envelope[0] * y[:env_len], envelope[1] * y[:env_len]])
+        else:
+            y_stereo = np.array(
+                [envelope[0][:y_len] * y, envelope[1][:y_len] * y])
+
+        return y_stereo, jams_out
+
+    @classmethod
+    def get_id(cls):
+        """Method to get the id of the pitch tracker
+
+        Returns
+        -------
+        str : 'guitar_strummer'
+        """
+        return 'guitar_strummer'
+
+    def _get_strum(self, start_t, end_t, chord, voicings, prev_voicing,
+                   backwards=False, velocity=None):
+        """
+        Parameters
+        ----------
+        start_t: float
+            starting time of the strum
+        end_t: float
+            ending time of the strum
+        chord: str
+            'Ab:maj6', 'D#:min', etc
+        voicings: dict
+            {chord_string: list of voicing}
```

Comment:
replace this line with
`keys are Chorde-Harte chord labels, values are lists of midi notes corresponding to different chord voicings`